packages/grant-explorer/src/features/contributors/__tests__/ViewContributionHistory.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { mockSigner } from "../../../test-utils";
+import { MemoryRouter } from "react-router-dom";
+
+const mockAddress = faker.finance.ethereumAddress();
+
+const mockTokens = {
+  ETH: {
+    name: "Ethereum",
+    chainId: "1",
+    address: "0x0000000000000000000000000000000000000000",
+    decimal: 18,
+    logo: "https://example.com/eth_logo.png",
+    default: true,
+    redstoneTokenId: "abc123",
+  },
+  DAI: {
+    name: "Dai",
+    chainId: "1",
+    address: "0x123456789abcdef",
+    decimal: 18,
+  },
+};
+
+const mockContributions = [
+  {
+    chainId: 1,
+    data: [
+      {
+        id: "1",
+        projectId: "project1",
+        roundId: "round1",
+        token: "ETH",
+        voter: "voter1",
+        grantAddress: faker.finance.ethereumAddress(),
+        amount: "10",
+        amountUSD: 100,
+        transaction: "transaction1",
+        roundName: "Round 1",
+        projectTitle: "Project 1",
+      },
+      {
+        id: "2",
+        projectId: "project2",
+        roundId: "round1",
+        token: "ETH",
+        voter: "voter2",
+        grantAddress: faker.finance.ethereumAddress(),
+        amount: "20",
+        amountUSD: 200,
+        transaction: "transaction2",
+        roundName: "Round 2",
+        projectTitle: "Project 2",
+      },
+    ],
+  },
+  {
+    chainId: 10,
+    data: [],
+  },
+];
+
+const useParamsFn = () => ({
+  address: mockAddress,
+});
+
+Object.defineProperty(window, "scrollTo", { value: () => {}, writable: true });
+
+jest.mock("../../common/Navbar");
+jest.mock("../../common/Auth");
+jest.mock("@rainbow-me/rainbowkit", () => ({
+  ConnectButton: jest.fn(),
+}));
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => jest.fn(),
+  useParams: useParamsFn,
+}));
+jest.mock("wagmi", () => ({
+  useSigner: () => mockSigner,
+  useEnsName: jest.fn().mockReturnValue({ data: "mockedEnsName" }),
+  useAccount: jest.fn().mockReturnValue({ data: "mockedAccount" }),
+}));
+
+jest.mock("ethereum-blockies", () => ({
+  __esModule: true,
+  default: {
+    create: jest.fn().mockReturnValue({
+      toDataURL: jest.fn().mockReturnValue("mockedAddressLogo"),
+    }),
+  },
+}));
+
+describe("<ViewContributionHistoryDisplay/>", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Should show donation impact & donation history", async () => {
+    render(
+      <MemoryRouter>
+        <ViewContributionHistoryDisplay
+          tokens={mockTokens}
+          contributions={mockContributions}
+          address={mockAddress}
+        />
+      </MemoryRouter>
+    );
+
+    expect(screen.getByText("Donation Impact")).toBeInTheDocument();
+    expect(screen.getByText("Donation History")).toBeInTheDocument();
+    expect(screen.getByTestId("profile-address")).toBeInTheDocument();
+    expect(screen.getByText("Share Profile")).toBeInTheDocument();
+    expect(screen.getByTestId("donation-history-table")).toBeInTheDocument();
```

Comment:
This test is not actually testing that the donatins are showing, instead of testing that the table is present, you can test that the contribution amounts are present, for example:

```
for(const contribtion of mockContributions) {
    expect(screen.getByText(contribution.transactionHash)).toBeInTheDocument();
}
```

it's just an example, i didn't test this code